receiver/k8sclusterreceiver/internal/ingress/ingress.go

```diff
@@ -0,0 +1,104 @@
+package ingress
+
+import (
+	"fmt"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/internal/common/maps"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/experimentalmetricmetadata"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	netv1 "k8s.io/api/networking/v1"
+	"strings"
+	"time"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/k8sclusterreceiver/internal/metadata"
+	imetadata "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/k8sclusterreceiver/internal/metadata"
+)
+
+const (
+	// Keys for ingress metadata.
+	AttributeK8SIngressUID  = "k8s.ingress.uid"
+	AttributeK8SIngressName = "k8s.ingress.name"
+	IngressCreationTime     = "ingress.creation_timestamp"
+)
+
+// Transform transforms the ingress to remove the fields.
+// IMPORTANT: Make sure to update this function before using new ingress fields.
+func Transform(r *netv1.Ingress) *netv1.Ingress {
+	newI := &netv1.Ingress{
+		ObjectMeta: metadata.TransformObjectMeta(r.ObjectMeta),
+	}
+	return newI
+}
+
+func RecordMetrics(mb *imetadata.MetricsBuilder, i *netv1.Ingress, ts pcommon.Timestamp) {
+	mb.RecordK8sIngressRuleCountDataPoint(ts, int64(len(i.Spec.Rules)))
+
+	rb := mb.NewResourceBuilder()
+	rb.SetK8sIngressUID(string(i.GetUID()))
+	rb.SetK8sIngressName(i.GetName())
+	rb.SetK8sClusterName("unknown")
+	rb.SetK8sIngressNamespace(i.GetNamespace())
+	rb.SetK8sIngressLabels(mapToString(i.GetLabels(), "&"))
+	rb.SetK8sIngressAnnotations(mapToString(i.GetAnnotations(), "&"))
+	rb.SetK8sIngressStartTime(i.GetCreationTimestamp().String())
+	rb.SetK8sIngressType("Ingress")
+	rb.SetK8sIngressRules(convertIngressRulesToString(i.Spec.Rules))
+	mb.EmitForResource(metadata.WithResource(rb.Emit()))
+}
+
+func mapToString(m map[string]string, seperator string) string {
+	var res []string
+	for k, v := range m {
+		res = append(res, fmt.Sprintf("%s=%s", k, v))
+	}
+	return strings.Join(res, seperator)
+}
+
+func convertIngressRulesToString(rules []netv1.IngressRule) string {
```

Comment:
Why do we need this function ? Can we just use https://pkg.go.dev/k8s.io/api/networking/v1#Ingress.String ?

Reply:
Tried `rule.String()` functionality, but it converts complex structs into a string format. This would make it hard for our developers to understand when displaying rules in a detailed view on the side-panel.

Right now:
host=www.example.com&http=(paths=(path=/foo&pathType=Prefix&backend=(service=(name=service1&port=(number=80))));host=api.example.com&http=(paths=(path=/&pathType=Prefix&backend=(service=(name=api-service&port=(number=8080))))

With `rule.String()` functionality:
&IngressRule{Host:www.example.com,IngressRuleValue:IngressRuleValue{HTTP:&HTTPIngressRuleValue{Paths:[]HTTPIngressPath{HTTPIngressPath{Path:/foo,Backend:IngressBackend{Resource:nil,Service:&IngressServiceBackend{Name:service1,Port:ServiceBackendPort{Name:,Number:80,},},},PathType:*Prefix,},},},},}&IngressRule{Host:api.example.com,IngressRuleValue:IngressRuleValue{HTTP:&HTTPIngressRuleValue{Paths:[]HTTPIngressPath{HTTPIngressPath{Path:/,Backend:IngressBackend{Resource:nil,Service:&IngressServiceBackend{Name:api-service,Port:ServiceBackendPort{Name:,Number:8080,},},},PathType:*Prefix,},},},},}

@tejaskokje-mw @bhogayatakb Which approach do you think is better?